nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -522,20 +528,18 @@ in {
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
       preStart = ''
-        if ! test -e /var/lib/matrix-synapse; then
-          mkdir -p /var/lib/matrix-synapse
-          chmod 700 /var/lib/matrix-synapse
-          chown -R matrix-synapse:matrix-synapse /var/lib/matrix-synapse
-          ${cfg.package}/bin/homeserver --config-path ${configFile} --keys-directory /var/lib/matrix-synapse/ --generate-keys
-        fi
+        ${cfg.package}/bin/homeserver \
+          --config-path ${configFile} \
+          --keys-directory /var/lib/matrix-synapse \
```

Comment:
Yeah, the [uid](https://github.com/NixOS/nixpkgs/blob/release-16.09/nixos/modules/misc/ids.nix#L245) and [gid](https://github.com/NixOS/nixpkgs/blob/release-16.09/nixos/modules/misc/ids.nix#L496) are both defined for this service. The change should not cause any issue for people already running Synapse as per discussion in #19986.